File: D_Content/Day8/D8_1_SampleService/app/src/main/java/org/techtown/service/MainActivity.java
package org.techtown.service;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText editText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editText = findViewById(R.id.editText);

        Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = editText.getText().toString();

                Intent intent = new Intent(getApplicationContext(), MyService.class);
                intent.putExtra("command", "show"); // 서비스로 보낼 인텐트 객체의 데이터 어떤 목적으로 ?
                intent.putExtra("name", name);

                startService(intent);
            }
        });

        Intent passedIntent = getIntent();
        processIntent(passedIntent); // 액티비티가 새로 만들어질 때 전달된 인텐트 처리
    }

    @Override
    protected void onNewIntent(Intent intent) {
        processIntent(intent); // 액티비티가 메모리에 있을 때 전달된 인텐트 처리

        super.onNewIntent(intent);
    }

    private void processIntent(Intent intent)
    {
        if (intent != null)
        {
            String command = intent.getStringExtra("command");
            String name = intent.getStringExtra("name");

            Toast.makeText(this, "command : " + command + ", name : " + name, Toast.LENGTH_SHORT).show();
        }
    }
}

File: D_About_Kotlin/03_AndroidByKotlin/07_ParcelableEx/app/src/main/java/org/techtown/parcelableex/MainActivity.kt
package org.techtown.parcelableex

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    val person = Person("Kildong", 30)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        butt_sub.setOnClickListener {

            val intent = Intent(this, SubActivity::class.java)
            var bundle = Bundle()
            bundle.putParcelable("selected_person", person)
            intent.putExtra("myBundle", bundle)
            startActivity(intent)
        }
    }
}